# Data Analysis on EA FIFA19
I am personally a FIFA fan since age of 16, I started with FIFA 15 on my iphone and can still remeber the face of Messi on the cover. Since then, I have played all EA's work on soccer from FIFA 15 to FIFA 20 and recenly I came across the FIFA 19 datasets and decided to find some info are helpful for myself. All the questions thats being asked in the analysis were came from my mind naturally as I diving into the dataset and hoping all info found hopefully can be helpful and practical to all FIFA fans and me myself.

## -Exploratory analysis

In [6]:

import pandas as pd
df = pd.read_csv('data.csv')
df

Unnamed: 0      ID                Name  Age  \
0               0  158023            L. Messi   31   
1               1   20801   Cristiano Ronaldo   33   
2               2  190871           Neymar Jr   26   
3               3  193080              De Gea   27   
4               4  192985        K. De Bruyne   27   
...           ...     ...                 ...  ...   
18202       18202  238813        J. Lundstram   19   
18203       18203  243165  N. Christoffersson   19   
18204       18204  241638           B. Worman   16   
18205       18205  246268      D. Walker-Rice   17   
18206       18206  246269           G. Nugent   16   

                                                Photo Nationality  \
0      https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1       https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2      https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3      https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4      https://cdn.sofifa.org/players/4/19/192985.png     Belgium   
...                                               ...         ...   
18202  https://cdn.sofifa.org/players/4/19/238813.png     England   
18203  https://cdn.sofifa.org/players/4/19/243165.png      Sweden   
18204  https://cdn.sofifa.org/players/4/19/241638.png     England   
18205  https://cdn.sofifa.org/players/4/19/246268.png     England   
18206  https://cdn.sofifa.org/players/4/19/246269.png     England   

                                      Flag  Overall  Potential  \
0      https://cdn.sofifa.org/flags/52.png       94         94   
1      https://cdn.sofifa.org/flags/38.png       94         94   
2      https://cdn.sofifa.org/flags/54.png       92         93   
3      https://cdn.sofifa.org/flags/45.png       91         93   
4       https://cdn.sofifa.org/flags/7.png       91         92   
...                                    ...      ...        ...   
18202  https://cdn.sofifa.org/flags/14.png       47         65   
18203  https://cdn.sofifa.org/flags/46.png       47         63   
18204  https://cdn.sofifa.org/flags/14.png       47         67   
18205  https://cdn.sofifa.org/flags/14.png       47         66   
18206  https://cdn.sofifa.org/flags/14.png       46         66   

                      Club  ... Composure Marking StandingTackle  \
0             FC Barcelona  ...      96.0    33.0           28.0   
1                 Juventus  ...      95.0    28.0           31.0   
2      Paris Saint-Germain  ...      94.0    27.0           24.0   
3        Manchester United  ...      68.0    15.0           21.0   
4          Manchester City  ...      88.0    68.0           58.0   
...                    ...  ...       ...     ...            ...   
18202      Crewe Alexandra  ...      45.0    40.0           48.0   
18203       Trelleborgs FF  ...      42.0    22.0           15.0   
18204     Cambridge United  ...      41.0    32.0           13.0   
18205      Tranmere Rovers  ...      46.0    20.0           25.0   
18206      Tranmere Rovers  ...      43.0    40.0           43.0   

       SlidingTackle GKDiving  GKHandling  GKKicking  GKPositioning  \
0               26.0      6.0        11.0       15.0           14.0   
1               23.0      7.0        11.0       15.0           14.0   
2               33.0      9.0         9.0       15.0           15.0   
3               13.0     90.0        85.0       87.0           88.0   
4               51.0     15.0        13.0        5.0           10.0   
...              ...      ...         ...        ...            ...   
18202           47.0     10.0        13.0        7.0            8.0   
18203           19.0     10.0         9.0        9.0            5.0   
18204           11.0      6.0         5.0       10.0            6.0   
18205           27.0     14.0         6.0       14.0            8.0   
18206           50.0     10.0        15.0        9.0           12.0   

      GKReflexes Release Clause  
0            8.0        €226.5M 

In [12]:
df.columns

Index(['Unnamed: 0', 'ID', 'Name', 'Age', 'Photo', 'Nationality', 'Flag',
       'Overall', 'Potential', 'Club', 'Club Logo', 'Value', 'Wage', 'Special',
       'Preferred Foot', 'International Reputation', 'Weak Foot',
       'Skill Moves', 'Work Rate', 'Body Type', 'Real Face', 'Position',
       'Jersey Number', 'Joined', 'Loaned From', 'Contract Valid Until',
       'Height', 'Weight', 'LS', 'ST', 'RS', 'LW', 'LF', 'CF', 'RF', 'RW',
       'LAM', 'CAM', 'RAM', 'LM', 'LCM', 'CM', 'RCM', 'RM', 'LWB', 'LDM',
       'CDM', 'RDM', 'RWB', 'LB', 'LCB', 'CB', 'RCB', 'RB', 'Crossing',
       'Finishing', 'HeadingAccuracy', 'ShortPassing', 'Volleys', 'Dribbling',
       'Curve', 'FKAccuracy', 'LongPassing', 'BallControl', 'Acceleration',
       'SprintSpeed', 'Agility', 'Reactions', 'Balance', 'ShotPower',
       'Jumping', 'Stamina', 'Strength', 'LongShots', 'Aggression',
       'Interceptions', 'Positioning', 'Vision', 'Penalties', 'Composure',
       'Marking', 'StandingTackle', 'SlidingT

In [41]:
# drop the Unnamed column 
df = df.loc[:, ~df.columns.str.contains('^Unnamed')]

In [42]:
df

ID                Name  Age  \
0      158023            L. Messi   31   
1       20801   Cristiano Ronaldo   33   
2      190871           Neymar Jr   26   
3      193080              De Gea   27   
4      192985        K. De Bruyne   27   
...       ...                 ...  ...   
18202  238813        J. Lundstram   19   
18203  243165  N. Christoffersson   19   
18204  241638           B. Worman   16   
18205  246268      D. Walker-Rice   17   
18206  246269           G. Nugent   16   

                                                Photo Nationality  \
0      https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1       https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2      https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3      https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4      https://cdn.sofifa.org/players/4/19/192985.png     Belgium   
...                                               ...         ...   
18202  https://cdn.sofifa.org/players/4/19/238813.png     England   
18203  https://cdn.sofifa.org/players/4/19/243165.png      Sweden   
18204  https://cdn.sofifa.org/players/4/19/241638.png     England   
18205  https://cdn.sofifa.org/players/4/19/246268.png     England   
18206  https://cdn.sofifa.org/players/4/19/246269.png     England   

                                      Flag  Overall  Potential  \
0      https://cdn.sofifa.org/flags/52.png       94         94   
1      https://cdn.sofifa.org/flags/38.png       94         94   
2      https://cdn.sofifa.org/flags/54.png       92         93   
3      https://cdn.sofifa.org/flags/45.png       91         93   
4       https://cdn.sofifa.org/flags/7.png       91         92   
...                                    ...      ...        ...   
18202  https://cdn.sofifa.org/flags/14.png       47         65   
18203  https://cdn.sofifa.org/flags/46.png       47         63   
18204  https://cdn.sofifa.org/flags/14.png       47         67   
18205  https://cdn.sofifa.org/flags/14.png       47         66   
18206  https://cdn.sofifa.org/flags/14.png       46         66   

                      Club                                       Club Logo  \
0             FC Barcelona    https://cdn.sofifa.org/teams/2/light/241.png   
1                 Juventus     https://cdn.sofifa.org/teams/2/light/45.png   
2      Paris Saint-Germain     https://cdn.sofifa.org/teams/2/light/73.png   
3        Manchester United     https://cdn.sofifa.org/teams/2/light/11.png   
4          Manchester City     https://cdn.sofifa.org/teams/2/light/10.png   
...                    ...                                             ...   
18202      Crewe Alexandra    https://cdn.sofifa.org/teams/2/light/121.png   
18203       Trelleborgs FF    https://cdn.sofifa.org/teams/2/light/703.png   
18204     Cambridge United   https://cdn.sofifa.org/teams/2/light/1944.png   
18205      Tranmere Rovers  https://cdn.sofifa.org/teams/2/light/15048.png   
18206      Tranmere Rovers  https://cdn.sofifa.org/teams/2/light/15048.png   

         Value   Wage  Special Preferred Foot  International Reputation  \
0      €110.5M  €565K     2202           Left                       5.0   
1         €77M  €405K     2228          Right                       5.0   
2      €118.5M  €290K     2143          Right                       5.0   
3         €72M  €260K     1471          Right                       4.0   
4        €102M  €355K     2281          Right                       4.0   
...        ...    ...      ...            ...                       ...   
18202     €60K    €1K     1307          Right                       1.0   
18203     €60K    €1K     1098          Right                       1.0   
18204     €60K    €1K     1189          Right                       1.0   
18205     €60K    €1K     1228          Right                       1.0   
18206     €60K    €1K     1321          Right                       1.0   

       Weak Foot  Skill Moves       Work Rate 

In [43]:
# find the value distribution among each column
df.describe()

ID           Age       Overall     Potential       Special  \
count   18207.000000  18207.000000  18207.000000  18207.000000  18207.000000   
mean   214298.338606     25.122206     66.238699     71.307299   1597.809908   
std     29965.244204      4.669943      6.908930      6.136496    272.586016   
min        16.000000     16.000000     46.000000     48.000000    731.000000   
25%    200315.500000     21.000000     62.000000     67.000000   1457.000000   
50%    221759.000000     25.000000     66.000000     71.000000   1635.000000   
75%    236529.500000     28.000000     71.000000     75.000000   1787.000000   
max    246620.000000     45.000000     94.000000     95.000000   2346.000000   

       International Reputation     Weak Foot   Skill Moves  Jersey Number  \
count              18159.000000  18159.000000  18159.000000   18147.000000   
mean                   1.113222      2.947299      2.361308      19.546096   
std                    0.394031      0.660456      0.756164      15.947765   
min                    1.000000      1.000000      1.000000       1.000000   
25%                    1.000000      3.000000      2.000000       8.000000   
50%                    1.000000      3.000000      2.000000      17.000000   
75%                    1.000000      3.000000      3.000000      26.000000   
max                    5.000000      5.000000      5.000000      99.000000   

           Crossing     Finishing  HeadingAccuracy  ShortPassing  \
count  18159.000000  18159.000000     18159.000000  18159.000000   
mean      49.734181     45.550911        52.298144     58.686712   
std       18.364524     19.525820        17.379909     14.699495   
min        5.000000      2.000000         4.000000      7.000000   
25%       38.000000     30.000000        44.000000     54.000000   
50%       54.000000     49.000000        56.000000     62.000000   
75%       64.000000     62.000000        64.000000     68.000000   
max       93.000000     95.000000        94.000000     93.000000   

            Volleys     Dribbling         Curve    FKAccuracy   LongPassing  \
count  18159.000000  18159.000000  18159.000000  18159.000000  18159.000000   
mean      42.909026     55.371001     47.170824     42.863153     52.711933   
std       17.694408     18.910371     18.395264     17.478763     15.327870   
min        4.000000      4.000000      6.000000      3.000000      9.000000   
25%       30.000000     49.000000     34.000000     31.000000     43.000000   
50%       44.000000     61.000000     48.000000     41.000000     56.000000   
75%       57.000000     68.000000     62.000000     57.000000     64.000000   
max       90.000000     97.000000     94.000000     94.000000     93.000000   

        BallControl  Acceleration   SprintSpeed       Agility     Reactions  \
count  18159.000000  18159.000000  18159.000000  18159.000000  18159.000000   
mean      58.369459     64.614076     64.726967     63.503607     61.836610   
std       16.686595     14.927780     14.649953     14.766049      9.010464   
min        5.000000     12.000000     12.000000     14.000000     21.000000   
25%       54.000000     57.000000     57.000000     55.000000     56.000000   
50%       63.000000     67.000000     67.000000     66.000000     62.000000   
75%       69.000000     75.000000     75.000000     74.000000     68.000000   
max       96.000000     97.000000     96.000000     96.000000     96.000000   

            Balance     ShotPower       Jumping       Stamina      Strength  \
count  18159.000000  18159.000000  18159.000000  18159.000000  18159.000000   
mean      63.966573     55.460047     65.089432     63.219946     65.311967   
std       14.136166     17.237958     11.820044     15.894741     12.557000   
min       16.000000      2.000000     15.000000     12.000000     17.000000   
25%       56.000000     45.000000     58.000000     56.000000     58.000000   
50%       66.000000     59.000000     66.000000     66.000000     67.000000   
75% 

In [44]:
# Have a really broad look at the correlation between each feature
df.corr()

ID       Age   Overall  Potential   Special  \
ID                        1.000000 -0.739208 -0.417025   0.047074 -0.231352   
Age                      -0.739208  1.000000  0.452350  -0.253312  0.236695   
Overall                  -0.417025  0.452350  1.000000   0.660939  0.606960   
Potential                 0.047074 -0.253312  0.660939   1.000000  0.383727   
Special                  -0.231352  0.236695  0.606960   0.383727  1.000000   
International Reputation -0.356191  0.253765  0.499491   0.372993  0.292208   
Weak Foot                -0.075784  0.059867  0.212015   0.162346  0.341855   
Skill Moves              -0.056914  0.027649  0.414463   0.354290  0.763412   
Jersey Number             0.182074 -0.241156 -0.218931  -0.010474 -0.133716   
Crossing                 -0.131994  0.130545  0.394972   0.246319  0.866417   
Finishing                -0.082323  0.068660  0.332515   0.243355  0.724244   
HeadingAccuracy          -0.106815  0.147183  0.340776   0.200988  0.644421   
ShortPassing             -0.136279  0.132894  0.502550   0.369189  0.906729   
Volleys                  -0.159915  0.142472  0.391338   0.254906  0.773974   
Dribbling                -0.030340  0.010166  0.372426   0.315019  0.874274   
Curve                    -0.169511  0.143276  0.419491   0.279944  0.851900   
FKAccuracy               -0.199549  0.193467  0.396892   0.230544  0.806414   
LongPassing              -0.186764  0.181310  0.483909   0.321437  0.846302   
BallControl              -0.100184  0.084969  0.460197   0.354396  0.912107   
Acceleration              0.133236 -0.158667  0.196869   0.234608  0.654337   
SprintSpeed               0.132437 -0.151682  0.210647   0.236771  0.645963   
Agility                  -0.019897 -0.019395  0.264952   0.222310  0.699673   
Reactions                -0.408617  0.453124  0.850045   0.513425  0.597169   
Balance                   0.048463 -0.089877  0.103160   0.138025  0.586788   
ShotPower                -0.166133  0.156947  0.441118   0.288318  0.835277   
Jumping                  -0.169369  0.177167  0.264435   0.109151  0.321846   
Stamina                  -0.053895  0.097793  0.365656   0.202563  0.792762   
Strength                 -0.259756  0.332798  0.349326   0.075769  0.192990   
LongShots                -0.161549  0.155096  0.420795   0.266740  0.840049   
Aggression               -0.228329  0.265190  0.395470   0.171174  0.666236   
Interceptions            -0.160602  0.197845  0.321326   0.154908  0.561676   
Positioning              -0.088330  0.082443  0.356493   0.245616  0.824307   
Vision                   -0.215170  0.187422  0.498894   0.348141  0.761992   
Penalties                -0.140657  0.139535  0.341429   0.224281  0.734533   
Composure                -0.384473  0.391023  0.727655   0.440008  0.752331   
Marking                  -0.110198  0.142817  0.286505   0.162801  0.561866   
StandingTackle           -0.085929  0.119745  0.252629   0.143564  0.538802   
SlidingTackle            -0.068409  0.103089  0.222811   0.128980  0.506968   
GKDiving                 -0.105594  0.101277 -0.025937  -0.053446 -0.674637   
GKHandling               -0.111149  0.106419 -0.025062  -0.054672 -0.673625   
GKKicking                -0.106652  0.104964 -0.029372  -0.059061 -0.670254   
GKPositioning            -0.118250  0.116402 -0.017674  -0.052589 -0.668272   
GKReflexes               -0.105778  0.103313 -0.023276  -0.053341 -0.673238   

                          International Reputation  Weak Foot  Skill Moves  \
ID                                       -0.356191  -0.075784    -0.056914   
Age                                       0.253765   0.059867     0.027649   
Overall                                   0.499491   0.212015     0.414463   
Potential                                 0.372993   0.162346     0.354290   
Special                                   0.292208   0.341855     0.763412   
International Reputation                  1.000000   0.128317     0.208153   
Weak Foot

In [47]:

print("The total entry we have:" , df.size)


The total entry we have: 1602216


In [50]:
print("The total registerd players:", df.shape[0])
print("The total measures of each player:", df.shape[1])

The total registerd players: 18207
The total measures of each player: 88


In [51]:
# Have a look at the null value distribution on the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18207 entries, 0 to 18206
Data columns (total 88 columns):
ID                          18207 non-null int64
Name                        18207 non-null object
Age                         18207 non-null int64
Photo                       18207 non-null object
Nationality                 18207 non-null object
Flag                        18207 non-null object
Overall                     18207 non-null int64
Potential                   18207 non-null int64
Club                        17966 non-null object
Club Logo                   18207 non-null object
Value                       18207 non-null object
Wage                        18207 non-null object
Special                     18207 non-null int64
Preferred Foot              18159 non-null object
International Reputation    18159 non-null float64
Weak Foot                   18159 non-null float64
Skill Moves                 18159 non-null float64
Work Rate                   18159 non-null 

In [52]:
# View first 5 rows of data with every columns
pd.set_option('display.max_columns', None)  
df.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo    Value  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  €110.5M   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png     €77M   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  €118.5M   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png     €72M   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png    €102M   

    Wage  Special Preferred Foot  International Reputation  Weak Foot  \
0  €565K     2202           Left                       5.0        4.0   
1  €405K     2228          Right                       5.0        4.0   
2  €290K     2143          Right                       5.0        5.0   
3  €260K     1471          Right                       4.0        3.0   
4  €355K     2281          Right                       4.0        5.0   

   Skill Moves       Work Rate   Body Type Real Face Position  Jersey Number  \
0          4.0  Medium/ Medium       Messi       Yes       RF           10.0   
1          5.0       High/ Low  C. Ronaldo       Yes       ST            7.0   
2          5.0    High/ Medium      Neymar       Yes       LW           10.0   
3          1.0  Medium/ Medium        Lean       Yes       GK            1.0   
4          4.0      High/ High      Normal       Yes      RCM            7.0   

         Joined Loaned From Contract Valid Until Height  Weight    LS    ST  \
0   Jul 1, 2004         NaN                 2021    5'7  159lbs  88+2  88+2   
1  Jul 10, 2018         NaN                 2022    6'2  183lbs  91+3  91+3   
2   Aug 3, 2017         NaN                 2022    5'9  150lbs  84+3  84+3   
3   Jul 1, 2011         NaN                 2020    6'4  168lbs   NaN   NaN   
4  Aug 30, 2015         NaN                 2023   5'11  154lbs  82+3  82+3   

     RS    LW    LF    CF    RF    RW   LAM   CAM   RAM    LM   LCM    CM  \
0  88+2  92+2  93+2  93+2  93+2  92+2  93+2  93+2  93+2  91+2  84+2  84+2   
1  91+3  89+3  90+3  90+3  90+3  89+3  88+3  88+3  88+3  88+3  81+3  81+3   
2  84+3  89+3  89+3  89+3  89+3  89+3  89+3  89+3  89+3  88+3  81+3  81+3   
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
4  82+3  87+3  87+3  87+3  87+3  87+3  88+3  88+3  88+3  88+3  87+3  87+3   

    RCM    RM   LWB   LDM   CDM   RDM   RWB    LB   LCB    CB   RCB    RB  \
0  84+2  91+2  64+2  61+2  61+2  61+2  64+2  59+2  47+2  47+2  47+2  59+2   
1  81+3  88+3  65+3  61+3  61+3  61+3  65+3  61+3  53+3  53+3  53+3  61+3   
2  81+3  88+3  65+3  60+3  60+3  60+3  65+3  60+3  47+3  47+3  47+3  60+3   
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
4  87+3  88+3  77+3  77+3  77+3  77+3  77+3  73+3  66+3  66+3  66+3  73+3   

   Crossing  Finishing  HeadingAccuracy  ShortPassing  Volleys  Dribbling  \
0      84.0       95.0             70.0          90.0     86.0       97.0   
1      84.0       94.0             89.0          81.0     

### A brief summary ( Descriptive & Diagnostic Analysis)
#### Geneal:
- there are 18207 soccer players registered in FIFA20 and there are 88 feature/measure for each player
- there seems a trend that order of players is based on the overall feature, which means better will be placed at front
- the data set has a relative high percentage of non-null values
- 

#### Players:
- Special feature has a deeper feature of player
- International reputation, weak foot, skill moves are measured on a scale of 5
- Info feature includes: ID, Name, Age, Photo, Nationality, Flag, Club, Club Logo, Wage, Value, International Reputation, Weak Foot, Skill Moves, Work Rate, Preferred Foor, Body Type,Position, Jersey Number, Joined, Loaned From, Contract Valid Until, Height, Weight, Release Clause
- Skill feature includes: Overall, Potential, Special, Skill points from LS - RB, Crossing - GKReflexes

#### Questions we can ask:
Before aksing questions, we wil decide the granularity for our analysis, which means the angle and level of details we are going to use.
- Player, Club, National Level (Country)



- For player: who is the best player? What builds a good plaer? What is the most important feature to be a good player? How does all feature contribute one player's value, wage, relase clause?
-


- For Team: Which team has most players? Most U19 players? Which team is the best? What makes them best? Which team has most tanlented player?
-


- For Country: Which country has most clubs? Which country has best league? What makes them? How does Country feature affect on Player and Club? Which country has most talented players?





